experiments/kompanion/cmd/export/export.go

```diff
@@ -280,25 +304,46 @@ func RunExport(ctx context.Context, opts *ExportOptions) error {
 		return fmt.Errorf("could not create %q directory: %w", reportTempDir, err)
 	}
 
+	shouldExcludeObject := func(id types.NamespacedName) bool {
+		if shouldExclude(id.Namespace, opts.ignoreNamespaces, opts.targetNamespaces) {
+			return true
+		}
+		return shouldExclude(id.Name, opts.ignoreObjects, opts.targetObjects)
+	}
```

Comment:
The logic is kind of twisted. I'm wondering if we can check GKNN (as a whole struct)  instead.

Reply:
Yes, the challenge is that we also want to check the namespace individually.  But yes, I think we can move in this direction!